pkg/playground/speed.js

```diff
@@ -176,12 +176,11 @@ function download(ev) {
 
     /* Allow use of HTTP URLs */
     if (path.value.indexOf("http") === 0) {
-        const anchor = document.createElement("a");
-        anchor.href = path.value;
+        const url = new URL(path.value);
         options.payload = "http-stream2";
-        options.address = anchor.hostname;
-        options.port = parseInt(anchor.port, 10);
-        options.path = anchor.pathname;
+        options.address = url.hostname;
+        options.port = parseInt(url.port, 10);
+        options.path = url.pathname;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18974-20230620-065303-a4180749-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)